src/hooks/useWidgetConfig.ts

```diff
@@ -0,0 +1,231 @@
+import {
+  TrackingAction,
+  TrackingCategory,
+  TrackingEventParameter,
+} from 'src/const/trackingKeys';
+import { useActiveTabStore } from 'src/stores/activeTab';
+import { useThemeStore } from 'src/stores/theme';
+import { useWidgetCacheStore } from 'src/stores/widgetCache';
+import type { LanguageKey } from 'src/types/i18n';
+import type { StarterVariantType } from 'src/types/internal';
+import { isIframeEnvironment } from 'src/utils/iframe';
+import { useConfig } from 'wagmi';
+import { useMemelist } from './useMemelist';
+import { useMultisig } from './useMultisig';
+import { useUserTracking } from './userTracking';
+
+interface UseWidgetConfigProps {
+  fromChain?: ChainId;
+  toChain?: ChainId;
+  fromToken?: string;
+  toToken?: string;
+  fromAmount?: string;
+  starterVariant?: StarterVariantType;
+  allowChains?: number[];
+  widgetIntegrator?: string;
+  customWidgetTheme?: WidgetTheme;
+}
+
+export const useWidgetConfig = ({
+  fromChain,
+  toChain,
+  fromToken,
+  toToken,
+  fromAmount,
+  starterVariant,
+  allowChains,
+  widgetIntegrator,
+  customWidgetTheme,
+}: UseWidgetConfigProps) => {
+  const widgetCache = useWidgetCacheStore((state) => state);
+  const [widgetTheme, configTheme] = useThemeStore((state) => [
+    state.widgetTheme,
+    state.configTheme,
+  ]);
+  const { activeTab } = useActiveTabStore();
+  const wagmiConfig = useConfig();
+  const { trackEvent } = useUserTracking();
+  const { i18n } = useTranslation();
+  const isGasVariant = activeTab === TabsMap.Refuel.index;
+  const integratorStringByType = useMemo(() => {
+    if (widgetIntegrator) {
+      return widgetIntegrator;
+    }
+    // all the trafic from mobile (including "/gas")
+    // if (!isDesktop) {
+    //   return process.env.NEXT_PUBLIC_INTEGRATOR_MOBILE;
+    // }
+    // all the trafic from web on "/gas"
+    if (isGasVariant) {
+      return process.env.NEXT_PUBLIC_WIDGET_INTEGRATOR_REFUEL;
+    }
+
+    return process.env.NEXT_PUBLIC_WIDGET_INTEGRATOR;
+  }, [widgetIntegrator, isGasVariant]) as string;
+  const { openWalletMenu } = useWalletMenu();
+  const partnerName = configTheme?.uid ?? 'default';
+  const { tokens: memeListTokens } = useMemelist({
+    enabled: partnerName === ThemesMap.Memecoins,
+  });
+  const { isMultisigSigner, getMultisigWidgetConfig } = useMultisig();
+  const { multisigWidget, multisigSdkConfig } = getMultisigWidgetConfig();
+  const allowedChainsByVariant = useMemo(
+    () =>
+      starterVariant === TabsMap.Refuel.variant
+        ? refuelAllowChains
+        : partnerName === ThemesMap.Memecoins
+          ? themeAllowChains
+          : [],
+    [starterVariant, partnerName],
+  );
+
+  // load environment config
+  const config: WidgetConfig = useMemo((): WidgetConfig => {
+    let rpcUrls = {};
+    try {
+      rpcUrls = {
+        ...JSON.parse(process.env.NEXT_PUBLIC_CUSTOM_RPCS),
+        ...publicRPCList,
+      };
+    } catch (e) {
+      if (process.env.DEV) {
```

Comment:
Dangerous there, DEV is not a standard env variable, error will be obfuscated all time and not throwing anything (making it harder to debug), just remove this if

Reply:
Can you explain?
![image](https://github.com/user-attachments/assets/4d938f34-3213-431b-91f4-25d0ae80d9fe)

Should we:
- remove the whole try-catch block?
- remove only the "if-condition"? ..or
- replace the env-var?

@tcheee @oktapodia

Reply:
Can you confirm and take another look if that is what you had in mind?

![image](https://github.com/user-attachments/assets/927df2c5-8c5c-4402-9a90-68209161afb8)